tools/Dashboard/DevHome.Dashboard/Views/DashboardView.xaml.cs

```diff
@@ -205,6 +214,19 @@ private async Task RestorePinnedWidgetsAsync()
         }
     }
 
+    private async Task DeleteAbandonedWidget(Widget widget)
+    {
+        var length = ViewModel.WidgetHostingService.GetWidgetHost()?.GetWidgets().Length;
+        Log.Logger()?.ReportInfo("DashboardView", $"Found abandoned widget, try to delete it...");
+        Log.Logger()?.ReportInfo("DashboardView", $"Before delete, {length} widgets for this host");
+
+        // If we have an abandoned widget, delete it.

```

Comment:
This comment is confusing.  Widget should always be not-null here.  If Widget is not null, how can the comment "If we have an..." be true if, at this point, we always have an abandoned widget?

Reply:
Comment moved and edited.